framework/Models/PostProcessors/BasicStatistics.py

```diff
@@ -594,27 +594,47 @@ def _computeWeightedPercentile(self,arrayIn,pbWeight,percent=0.5):
       Method to compute the weighted percentile in a array of data
       @ In, arrayIn, list/numpy.array, the array of values from which the percentile needs to be estimated
       @ In, pbWeight, list/numpy.array, the reliability weights that correspond to the values in 'array'
-      @ In, percent, float, the percentile that needs to be computed (between 0.01 and 1.0)
-      @ Out, result, float, the percentile
+      @ In, percent, float/list/numpy.array, the percentile(s) that needs to be computed (between 0.01 and 1.0)
+      @ Out, result, float/list, the percentile(s)
     """
 
+    # only do the argsort once for all requested percentiles
     idxs                   = np.argsort(np.asarray(list(zip(pbWeight,arrayIn)))[:,1])
     # Inserting [0.0,arrayIn[idxs[0]]] is needed when few samples are generated and
     # a percentile that is < that the first pb weight is requested. Otherwise the median
     # is returned.
     sortedWeightsAndPoints = np.insert(np.asarray(list(zip(pbWeight[idxs],arrayIn[idxs]))),0,[0.0,arrayIn[idxs[0]]],axis=0)
     weightsCDF             = np.cumsum(sortedWeightsAndPoints[:,0])
+    if not hasattr(percent, '__iter__'):
+      # percent is int or float
+      result = self._computeSingleWeightedPercentile(percent, weightsCDF, sortedWeightsAndPoints)
+    else:
+      # percent is iterable
+      result = [self._computeSingleWeightedPercentile(pct, weightsCDF, sortedWeightsAndPoints) for pct in percent]
+
+    return result
```

Comment:
"_computeSingleWeightedPercentile" is there to avoid copy/pasting the same code in the if..else.. statement. If I had written this originally, I would have used numpy.interp which would take float, list, or array and would have avoided writing "_computeSingleWeightedPercentile" and the if..else.. statement. The only reason I did not implement this is that it changes the way the percentiles are calculated (linear interpolation instead of the midpoint between the two nearest data points) and would require updating many gold files. I think numpy.interp would speed the calculation because the for loop would be eliminated. 

I can change the "_computeWeightedPercentile" function to accept a list of percents instead of the accepting float or list. If you think it would be worth it to do the numpy.interp method, I could try that as well.

Reply:
"_computeWeightedPercentile" has been changed to accept a list of percents. With the new refactoring, "_computeSingleWeightedPercentile" is no longer necessary.